modules/apps/frontend-icons/frontend-icons-web/src/main/resources/META-INF/resources/js/IconConfiguration.js

```diff
@@ -0,0 +1,297 @@
+				});
+
+				const newIcons = {...icons};
+
+				delete newIcons[iconPackName];
+
+				setIcons(newIcons);
+			}
+		);
+	};
+
+	return (
+		<ClayLayout.Sheet>
+			<ClayLayout.ContentRow className="mb-5" containerElement="h2">
+				<ClayLayout.ContentCol containerElement="span" expand>
+					{Liferay.Language.get('frontend-icons-configuration-name')}
+				</ClayLayout.ContentCol>
+			</ClayLayout.ContentRow>
+
+			<h4>{Liferay.Language.get('icon-packs')}</h4>
+
+			<label className="form-control-label">
+				<span className="form-control-label-text">
+					{Liferay.Language.get('search-icons')}
+				</span>
+
+				<ClayInput
+					onChange={(event) => setSearchQuery(event.target.value)}
+					placeholder={Liferay.Language.get('search-icons')}
+					type="text"
+					value={searchQuery}
+				/>
+			</label>
+
+			<ClayPanel.Group className="mt-4">
+				{iconPackNames.map((iconPackName) => (
+					<div className="d-flex" key={iconPackName}>
+						<ClayPanel
+							collapsable
+							displayTitle={`${iconPackName} (${filteredIcons[iconPackName]?.icons.length})`}
+							displayType="secondary"
+							showCollapseIcon={true}
+							style={{flex: 1}}
+						>
+							<ClayPanel.Body className="list-group-card">
+								{filteredIcons[iconPackName].editable && (
+									<ClayDropDownWithItems
+										items={[
+											{
+												label:
+													'Add Icons Pack from spritemap',
+												onClick: () =>
+													setAddModal({
+														existingIconPackName: iconPackName,
+														uploadSpritemap: true,
+														visible: true,
+													}),
+											},
+											{
+												label:
+													'Add Icons Pack from existing icons',
+												onClick: () =>
+													setAddModal({
+														existingIconPackName: iconPackName,
+														uploadSpritemap: false,
+														visible: true,
+													}),
+											},
+										]}
+										trigger={
+											<ClayButton
+												className="mt-2"
+												displayType="secondary"
+												small
+											>
+												Add Icons
+											</ClayButton>
+										}
+									/>
+								)}
+
+								<ul className="list-group">
+									{filteredIcons[iconPackName].icons
+										.sort()
+										.map((icon) => (
+											<li
+												className="list-group-card-item w-25"
+												key={icon.name}
+											>
+												<ClayButton
+													displayType={null}
+													onClick={() => {
+														if (
+															filteredIcons[
+																iconPackName
+															].editable
+														) {
+															setDeleteModal({
+																iconPackName,
+																selectedIcon:
+																	icon.name,
+																visible: true,
+															});
+														}
+													}}
+												>
+													<ClayIcon
+														spritemap={`/o/icons/${iconPackName}.svg?${referenceTime}`}
+														symbol={icon.name}
```

Comment:
Since the JS API to grab icons has been extracted to another task, I had to hardcode the url here